lib/vault/rails.rb

```diff
@@ -192,12 +200,27 @@ def self.force_encoding(str)
 
     private
 
-    def self.log_warning
-      if defined?(::Rails) && ::Rails.logger != nil
-        ::Rails.logger.warn do
-          "[vault-rails] Using in-memory cipher - this is not secure " \
-          "and should never be used in production-like environments!"
+    def self.with_retries(client = self.client, &block)
+      exceptions = [Vault::HTTPConnectionError, Vault::HTTPServerError]
+      options = {
+        attempts: self.retry_attempts,
+        base:     self.retry_base,
+        max_wait: self.retry_max_wait,
+      }
+
+      client.with_retries(exceptions, options) do |i, e|
+        if !e.nil?
+          log_warning "[vault-rails] (#{i}) An error occurred when trying to "
+            "communicate with Vault: #{e.message}"
         end
```

Comment:
The block will still yield for the first request, but the exception will be nil. So the first request isn't actually a retry